packages/core/src/types.ts

```diff
@@ -59,12 +59,56 @@ export interface Position {
 // Config
 //------------------------------------------------------------------------------
 
-export type RuleSeverity = 0 | 1 | 2 | "off" | "warn" | "error";
+/**
+ * The human readable severity level used in a configuration.
+ */
+export type SeverityName = "off" | "warn" | "error";
+
+/**
+ * The numeric severity level for a rule.
+ *
+ * - `0` means off.
+ * - `1` means warn.
+ * - `2` means error.
+ *
+ */
+export type SeverityLevel = 0 | 1 | 2;
+
+/**
+ * Represents the configuration options for the core linter.
+ */
+export interface LinterOptionsConfig {
+	/**
+	 * Indicates whether or not inline configuration is evaluated.
+	 */
+	allowInlineConfig?: boolean;
+
+	/**
+	 * Indicates what to do when an unused disable directive is found.
+	 */
+	reportUnusedDisableDirectives?: boolean | SeverityName;
+}
```

Comment:
These are options passed to `Linter#verify()` ([`VerifyOptions`](https://github.com/eslint/eslint/blob/7bcda760369c44d0f1131fccaaf1ccfed5af85f1/lib/linter/linter.js#L107-L118)), not `linterOptions` in config objects?

Reply:
No, this is for the config object. What am I missing?

Reply:
In config objects (e.g., in an `eslint.config.js` file), [`linterOptions`](https://github.com/eslint/eslint/blob/7bcda760369c44d0f1131fccaaf1ccfed5af85f1/lib/config/flat-config-schema.js#L555-L559) objects have properties:

* `noInlineConfig`: allowed values are `true`, `false`.
* `reportUnusedDisableDirectives`: allowed values are `"error"`, `"warn"`, `"off"`, `2`, `1`, `0`, and also legacy `true`, `false`. https://github.com/eslint/eslint/blob/7bcda760369c44d0f1131fccaaf1ccfed5af85f1/lib/config/flat-config-schema.js#L287-L305

Reply:
Ah, so just missed the numbers. Thanks.